custom/icds/management/commands/contact_phone_fix.py

```diff
@@ -0,0 +1,174 @@
+import csv
+import six
+import sys
+import time
+
+from datetime import datetime
+from xml.etree import cElementTree as ElementTree
+
+from django.core.management.base import BaseCommand
+
+from corehq.apps.users.util import SYSTEM_USER_ID
+from corehq.form_processor.backends.sql.dbaccessors import CaseReindexAccessor
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.util.log import with_progress_bar
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.hqcase.utils import submit_case_blocks
+from corehq.form_processor.backends.sql.dbaccessors import iter_all_rows
+
+from dimagi.utils.chunked import chunked
+from casexml.apps.case.mock import CaseBlock
+
+DOMAIN = "icds-cas"
+CASE_TYPE = "person"
+
+PHONE_NUMBER_PROPERTY = "contact_phone_number"
+HAS_MOBILE_PROPERTY = "has_mobile"
+HAS_MOBILE_PROPERTY_VALUE = "yes"
+CONTACT_PHONE_NUMBER_IS_VERIFIED = "contact_phone_number_is_verified"
+CONTACT_PHONE_NUMBER_IS_VERIFIED_VALUE = 1
+CASE_ITERATION_COUNT = 10000
+MAX_RESCUE_EXCEPTIONS_ON_UPDATE = 5
+CSV_HEADERS = ['Case Id']
+
+TEST_STATES = []
+
+for loc in SQLLocation.active_objects.filter(location_type__code='state', domain=DOMAIN):
+    if loc.metadata.get('is_test_location') == 'test':
+        TEST_STATES.append(loc.name)
+
+
+class Command(BaseCommand):
+    help = """
+    Find the person cases which are
+        - not deleted
+        - not belonging to test locations,
+        - with has_mobile case_property set to "yes",
+            - if the contact_phone_number_is_verified is 1, 
+            update the contact_phone_number property by appending 91 with phone_number,
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(Command, self).__init__(*args, **kwargs)
+        self.db_alias = None
+        self.log_progress = False
+        self.test_locations = None
+        self.case_accessor = CaseAccessors(DOMAIN)
+
+    def add_arguments(self, parser):
+        parser.add_argument('db_alias')
+        parser.add_argument(
+            '--log',
+            action='store_true',
+            dest='log_progress',
+            default=False,
+            help="log progress"
+        )
+
+    def _case_needs_to_be_updated(self, case):
+        if case.deleted:
+            return False
+        assert case.type == CASE_TYPE
+        if bool(case.owner_id) and case.owner_id in self.test_locations:
+            return False
+        if case.get_case_property(HAS_MOBILE_PROPERTY) == HAS_MOBILE_PROPERTY_VALUE:
+            phone_number = case.get_case_property(PHONE_NUMBER_PROPERTY)
+            phone_number_verified = case.get_case_property(CONTACT_PHONE_NUMBER_IS_VERIFIED)
+            if len(phone_number) == 10 and phone_number_verified == CONTACT_PHONE_NUMBER_IS_VERIFIED_VALUE:
```

Comment:
Confirming that this will only update cases that don't have 91 appended already?

I am guessing you will find some faulty numbers in there, for ex: 91 followed by 8 digits when someone entered a 8 digit phone number at the first place. the checks on phone number length might have been added later. But sounds okay to ignore that.